### PR TITLE
Fix issue #260: smarter rendering of array elements across lines

### PR DESCRIPTION
**Pull request by dnkolegov**
The fix is based on discussion and  suggestion in https://github.com/estools/escodegen/issues/260.

### CODE REVIEW

File: escodegen.js [hunk 1]
```diff
@@ -2056,7 +2056,7 @@
             if (!expr.elements.length) {
                 return '[]';
             }
-            multiline = isPattern ? false : expr.elements.length > 1;
+            multiline = isPattern ? false : expr.elements.length > 10 && expr.elements[0].type !== Syntax.Literal;
```
Comment (michaelficarra) [inline escodegen.js]:
```js
!isPattern && expr.elements.length > 10 && expr.elements[0].type !== Syntax.Literal
```

**Comment by michaelficarra** [comment]
I don't like this approach. It is a rough approximation which may be very incorrect for certain inputs. The solution I suggested in #260 would actually solve this. :-1:

**Comment by dnkolegov** [comment]
Hi @michaelficarra. Thanks for suggestion.

> this can be done by determining the formatting based on the size/content of the generated source text for each element

Could you make clear what do you mean here? Using of range and loc modes for esprima?

File: escodegen.js (hunk 1 above)
Comment (dnkolegov) [inline escodegen.js]: Do you mean this?
```
multiline = isPattern ? false : !isPattern && expr.elements.length > 10 && expr.elements[0].type !== Syntax.Literal;
```
`isPattern` in this is `false` and `!isPattern` is `true`.

**Comment by michaelficarra** [comment]
@dnkolegov The suggestion I had was to split across lines based on the length of the result of calling `generateExpression` on the array elements. Closing this, as it's not making a decision based on a useful characteristic of the array.

**Comment by healzer** [comment]
> 
> 
> @dnkolegov The suggestion I had was to split across lines based on the length of the result of calling `generateExpression` on the array elements. Closing this, as it's not making a decision based on a useful characteristic of the array.

It would be nice to add an API option for this feature. I need this implemented :p

**Comment by michaelficarra** [comment]
@healzer Feel free to implement it and send a PR.

**Comment by healzer** [comment]
> 
> 
> @healzer Feel free to implement it and send a PR.

Maybe I will in the future. For now I locally set the "multiline" to false. :)

**Pull request closed without merging**